src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java

```diff
@@ -213,4 +213,24 @@ public void testStartCommand() throws SaltStackException {
         assertEquals(job.getJid(), "20150211105524392307");
         assertEquals(job.getMinions(), Arrays.asList("myminion"));
     }
+    
+    @Test
+    public void testStartCommandAsync() throws SaltStackException {
+    	 stubFor(post(urlEqualTo("/minions")).willReturn(
+                 aResponse().withStatus(HttpURLConnection.HTTP_OK)
+                         .withHeader("Content-Type", "application/json")
+                         .withBody(JSON_START_COMMAND_RESPONSE)));
+
+        Future<Job> future = client.startCommandAsync("*", "test.ping", null, null);
+        Job job;
+        try {
+        	job = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new SaltStackException(e);
```

Comment:
Sorry @gborp, I didn't notice other methods were doing the same. I would say, let's merge this and then have another PR where we remove all needless `Exception` wrapping (option A).

In my opinion, tests should throw `Exception` - listing specific subclasses is just maintenance burden.